docs/okeps/okep-4380-network-qos.md

```diff
@@ -0,0 +1,473 @@
+
+The workload taffic can be either east west (pod to pod traffic) or north south traffic (pod to external traffic) types in a Kubernetes cluster which is limited by finite bandwidth. So NetworkQoS must ensure high priority applications get the necessary NetworkQoS marking so that it can prevent network conjestion.
+
+## Goals
+
+- Provide a mechanism for users to set DSCP (Differentiated Services Code Point) marking on east west and north south traffic.
+- Provide a mechanism for users to set Metering on east west and north south traffic.
+
+## Non-Goals
+
+- Ingress Network QoS.
+
+- Consolidating with current `kubernetes.io/egress-bandwidth` and `kubernetes.io/ingress-bandwidth` annotations.
+Nonetheless, the work done here does not interfere with the current bandwidth NetworkQoS mechanism.
+
+- The DSCP marking does not need to be handled or acted upon by OVN-Kubernetes, just added to selected headers.
+
+## User-Stories/Use-Cases
+#### Story 1
+
+As an OVN-Kubernetes user, I want to configure DSCP values for both east-west and north-south traffic.
+
+For instance, in a datacenter, pods may need to differentiate traffic between storage operations and streaming services. To achieve true end-to-end traffic differentiation, packets must be marked at the chassis where the pod is running.
+
+Using the `inherit` setting of OVS, DSCP markings are applied to the outer packet. These markings influence NIC behavior, which enforces traffic policing. Subsequently, the same markings are used by network switches in the underlay to apply additional policies.
+
+Beyond QoS, DSCP markings enable other use cases. For example, applications like gateways can treat packets differently based on these markings, offering tailored processing or forwarding behavior.
+
+This feature is not limited to QoS; it also allows for enhanced packet forwarding behavior, such as using `iptables` on a gateway to control traffic based on DSCP marks.
+
+For more details on this use case, check out [this presentation recorded at KubeCon 2024](https://youtu.be/76tfwAvDWX8?feature=shared).
+
+```text
+                                             .---------------.
++---------+                              _.-'                 `--.
+|       +-+---+    +------+   +----+    /         overlay         \
+|   Pod |eth0 +----+br-int+---|NIC |--->       over network        )
+|       +-+---+    +------+   +----+    `.        fabric         ,'
++---------+                               `--.               _.-'
+                                              `-------------'
+```
+
+#### Story 2
+
+As a user of OVN-Kubernetes, I want to apply bandwidth limit (rate and burst) for east west and north south traffic.
```

Comment:
Here an example with different applications would make sense, so you can describe a specific setup, in which it makes sense to apply bandwidth limit, and again whether there is any difference between namespace owner vs cluster admin. e.g. I can probably imagine why cluster admin would want to set burst limit per namespace, so that different namespaces don't affect shared network too much, but I am not sure of a use case for namespace owner. 
"so that different namespaces don't affect shared network too much" from my example is the WHY (which still could use some more details or examples), we need to explain the WHY for your real use cases